### PR TITLE
MB-10430 display info in move task order tab shipment sidebar

### PR DESCRIPTION
**Pull request by YanZ777**
## Summary

Updated the move details sidebar to show the Facility info and address, the Service order number, and the accounting codes.
The Edit links just take the user back to the moves queue. This is intentional as the actual linking will occur in [MB-10437](https://dp3.atlassian.net/browse/MB-10437) and [MB-1035](https://dp3.atlassian.net/browse/MB-10435)

`ShipmentDetailsSidebar` got changed to just take in a whole shipment.  The amount of props was getting to be a bit ridiculous and it was effectively passing down a shipment anyways with how the addresses worked.
I've upgraded one of the shipments in the devseed data to have a storage facility, service order number, and accounting codes. 
I also made the decision to rename all the things to `ntsTac` and `ntsSac` to match what the backend is providing.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

(You may have to reseed the database before going through these instructions.)
1. Access the office site.
2. Login as a TOO
3. Use the move code `HGNTSR`
4. Select the "Move task order" tab and scroll down to the NTS-Release shipment.
5. There should be new information on the side.

1. Alternatively, you could go to your local run of storybook and access [Shipment Details](http://localhost:6006/?path=/story/office-components-shipment-details--default)

## Verification Steps for Author

These are to be checked by the author.

- [ x ] Request review from a member of a different team.
- [ x ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output

### On the site:
image

### Storybook: 
image

### CODE REVIEW

File: src/components/Office/ShipmentDetails/ShipmentDetailsSidebar.jsx [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 import SimpleSection from 'containers/SimpleSection/SimpleSection';
-import { AddressShape } from 'types/address';
-import { AgentShape } from 'types/agent';
-import { formatAgent, formatAddress } from 'utils/shipmentDisplay';
+import { retrieveSAC, retrieveTAC, formatAgent, formatAddress, formatAccountingCode } from 'utils/shipmentDisplay';
+import { ShipmentShape } from 'types/shipment';
+import { OrdersLOAShape } from 'types/order';
+
+const ShipmentDetailsSidebar = ({ className, shipment, ordersLOA }) => {
+  const { agents, secondaryAddresses, serviceOrderNumber, storageFacility, sacType, tacType } = shipment;
```
Comment (donaldthai) [inline src/components/Office/ShipmentDetails/ShipmentDetailsSidebar.jsx:11]: I'm not seeing the agents being populated. Seems like the name should be `mtoAgents`.

Comment (YanZ777) [reply src/components/Office/ShipmentDetails/ShipmentDetailsSidebar.jsx]: Good catch! I've updated it. (There hasn't been any in the devseed, do you want me to add some?)

File: src/components/Office/ShipmentDetails/ShipmentDetails.jsx [hunk 2]
```diff
@@ -17,6 +17,13 @@ const ShipmentDetails = ({
   handleSubmitSITExtension,
 }) => {
   const { originDutyStation, destinationDutyStation, entitlement } = order;
+  const ordersLOA = {
+    tac: order.tac,
+    sac: order.sac,
+    ntsTac: order.ntsTac,
+    ntsSac: order.ntsSac,
+  };
+
```
Comment (kylehilltruss) [inline src/components/Office/ShipmentDetails/ShipmentDetails.jsx:26]: I spent way too much time reading the object destructuring pages on MDN and trying to figure out how to optimize this, unsuccessfully.

**Review by kylehilltruss — APPROVED** [review]
Functionality works and looks good; code changes make sense. Thank you for the devseed move!

**Review by amanda5iu — COMMENTED** [review]
I got a blank sidebar when running this locally, but the details appear in Storybook. Can we show lot # in the facility address as well, so that people know what it looks like when there is one?

Also, the second screenshot you shared shows a releasing agent on the NTS-release, but that doesn't apply to this shipment type. Not sure if it will always appear like this?

File: src/components/Office/ShipmentDetails/ShipmentDetails.jsx (hunk 2 above)
Comment (YanZ777) [reply src/components/Office/ShipmentDetails/ShipmentDetails.jsx:26]: Thanks for looking! I couldn't find one either, unfortunately.

**Comment by YanZ777** [comment]
> I got a blank sidebar when running this locally, but the details appear in Storybook. Can we show lot # in the facility address as well, so that people know what it looks like when there is one?
> 
> Also, the second screenshot you shared shows a releasing agent on the NTS-release, but that doesn't apply to this shipment type. Not sure if it will always appear like this?

You might have to repopulate your devseed data (I'll add that to the instructions). 
I added in the lot number, how did you want it formatted relative to the address? 
As for showing releasing agent, that's mostly me putting as much information into the sidebar display as possible. Do you prefer I take it out?

**Comment by amanda5iu** [comment]
> You might have to repopulate your devseed data (I'll add that to the instructions).
I added in the lot number, how did you want it formatted relative to the address?
As for showing releasing agent, that's mostly me putting as much information into the sidebar display as possible. Do you prefer I take it out?

@YanZ777 I ran `make db_dev_e2e_populate` and can now see the sidebar!  Some notes:
- The `Edit` links seem to be inheriting the styling of the data label. It should use the `a small` link style.
- Accounting codes should say whether it's a TAC or SAC, like `TAC: 1234 (HHG)`. [Zeplin component for reference](https://zpl.io/aB3n84A)
- NTS-releases should show Receiving Agent, but they'll never have a releasing agent
- The reverse is true for NTS - there's a releasing agent, but no receiving.

For the lot #, it can just be `Lot [#]` on a new line. There's an example in [Zeplin](https://zpl.io/aB3n84A) - see the Preview page version of the NTS-release card.

**Comment by YanZ777** [comment]
> @YanZ777 I ran `make db_dev_e2e_populate` and can now see the sidebar! Some notes:
> 
> * The `Edit` links seem to be inheriting the styling of the data label. It should use the `a small` link style.
> * Accounting codes should say whether it's a TAC or SAC, like `TAC: 1234 (HHG)`. [Zeplin component for reference](https://zpl.io/aB3n84A)
> * NTS-releases should show Receiving Agent, but they'll never have a releasing agent
> * The reverse is true for NTS - there's a releasing agent, but no receiving.
> 
> For the lot #, it can just be `Lot [#]` on a new line. There's an example in [Zeplin](https://zpl.io/aB3n84A) - see the Preview page version of the NTS-release card.


Ok I think I should have addressed all of these. 
I've updated the devseed to take out the releasing agent.
I left the agents as is for storybook Since I want to be able to see all the sidebar options there are.

**Review by donaldthai — APPROVED** [review]
🤩

**Review by amanda5iu — APPROVED** [review]
Looks great!!

**Pull request merged**